packages/remix-react/data.ts

```diff
@@ -35,23 +36,143 @@ export async function fetchData(
   routeId: string,
   signal: AbortSignal,
   submission?: Submission
-): Promise<Response | Error> {
+): Promise<[Response | Error, any]> {
```

Comment:
Can we reuse the same data structure we're using on the backend here? i.e. `DeferredResponse`?